### DISCUSSION TITLE
Email OTP Code expires after progammatically updating user's email_confirm boolean

### DISCUSSION THREAD

**Discussion by jacobjohn-transitiv** [supabase/supabase#34841] [Troubleshooting]
# Bug report

I noticed that if a user does not sign into our application within 24 hours of receiving the initial invite email (initiated by auth.admin.inviteUserByEmail in Remix.js), their account becomes invalid and we would have to remove the email from the user table and re-invite them. 
To prevent this from happening, we found out that we can programmatically update the newly created user's email_confirm boolean to true, which prevents the expiry of the account after 24 hours. However, we noticed that the initial OTP code that is sent in the original invite email automatically expires. 

Is the expiry of the OTP code after updating the user's email_confirm a bug? If it is not, what would be the correct way to invite a user by email and not have their account expire/invalidated after 24 hours? Below is the relevant code for sending an invite by email, and updating the user's email_confirm.

```
  const { data, error } = await adminSupabase.auth.admin.inviteUserByEmail(
    email,
    {
      data: {
        ...userMetaData,
        userId: userData.user.id,
        email,
      } as InviteUserMetaData,
    }
  )

  if (data?.user) {
    const { data: updatedUserData, error: updatedUserError } =
      await adminSupabase.auth.admin.updateUserById(data.user.id, {
        email_confirm: true,
      })
    return { data: updatedUserData, error: updatedUserError }
  }
  ```

**Comment by encima** [comment]
Hey there, 
thanks for opening! The email expiry after 24 hours is expected behaviour.
The best way to work with this is to detect the expiry of the link (from the error in the URL or `invite_not_found`) and then you can resend the link proactively using [this](https://supabase.com/docs/reference/javascript/auth-admin-inviteuserbyemail)

Does that work for you?

**Comment by jacobjohn-transitiv** [comment]
The issue here is that when I try to re-invite the user, it says the email is already taken. I would have to remove the user from the user table, then re-invite them. Is making the user's entire account invalid expected behavior? Not sure why they can't request a new token.

**Comment by andreicos1** [comment]
Have you found a solution to this?

**Comment by RatnadeepRoy01** [comment]
Hey if this is a real bug i would like to solve this hope maintainer can clarify this

**Comment by jacobjohn-transitiv** [comment]
I have not figured out a fix for this. Our current "solution" is to update the user's `email_confirmed ` bool to true after sending the initial email invite. This will prevent the 24 hour account expiry. The issue with this is that the initial OTP code sent in the initial email invite is then void/invalid, so they would have to request a new OTP code before they can sign into the app.

One workaround to this is to send two emails - one explaining the invite and to expect a separate email with a fresh OTP code. A bit cumbersome, but this is the only solution we could come up with.

**Comment by Hallidayo** [comment]
Hi everyone, due to inactivity on this issue I've moved the issue over to discussions/enhancements.
